From: Andrew Morton <akpm@linux-foundation.org>
Date: Thu, 24 Apr 2025 17:15:43 -0700
Subject: [PATCH] 
 mm-selftests-add-a-test-to-verify-mmap_changing-race-with-eagain-checkpatch-fixes

WARNING: do not add new typedefs
#86: FILE: tools/testing/selftests/mm/uffd-unit-tests.c:1284:
+typedef enum {

WARNING: braces {} are not necessary for single statement blocks
#106: FILE: tools/testing/selftests/mm/uffd-unit-tests.c:1304:
+	if (!fp) {
+		return THR_STATE_UNKNOWN;
+	}

WARNING: Use of volatile is usually wrong: see Documentation/process/volatile-considered-harmful.rst
#135: FILE: tools/testing/selftests/mm/uffd-unit-tests.c:1333:
+	volatile pid_t *pid = opaque;

ERROR: trailing statements should be on next line
#154: FILE: tools/testing/selftests/mm/uffd-unit-tests.c:1352:
+	while (uffd_read_msg(fd, &msg));

WARNING: Use of volatile is usually wrong: see Documentation/process/volatile-considered-harmful.rst
#180: FILE: tools/testing/selftests/mm/uffd-unit-tests.c:1378:
+	while (!(volatile pid_t)pid)

total: 1 errors, 4 warnings, 215 lines checked

NOTE: For some of the reported defects, checkpatch may be able to
      mechanically convert to the typical style using --fix or --fix-inplace.

./patches/mm-selftests-add-a-test-to-verify-mmap_changing-race-with-eagain.patch has style problems, please review.

NOTE: If any of the errors are false positives, please report
      them to the maintainer, see CHECKPATCH in MAINTAINERS.

Please run checkpatch prior to sending patches

Cc: Peter Xu <peterx@redhat.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---
 tools/testing/selftests/mm/uffd-unit-tests.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/tools/testing/selftests/mm/uffd-unit-tests.c b/tools/testing/selftests/mm/uffd-unit-tests.c
index e3b5046d02d8..c73fd5d455c8 100644
--- a/tools/testing/selftests/mm/uffd-unit-tests.c
+++ b/tools/testing/selftests/mm/uffd-unit-tests.c
@@ -1301,9 +1301,8 @@ static thread_state thread_state_get(pid_t tid)
 	snprintf(tmp, sizeof(tmp), "/proc/%d/status", tid);
 	fp = fopen(tmp, "r");
 
-	if (!fp) {
+	if (!fp)
 		return THR_STATE_UNKNOWN;
-	}
 
 	while (fgets(tmp, sizeof(tmp), fp)) {
 		p = strstr(tmp, header);
-- 
2.39.5

